coul.c

```diff
@@ -2916,7 +2918,13 @@ int main(int argc, char **argv, char **envp) {
 
     init_post();
     report_init(stdout, argv[0]);
+    if (freopen(rpath, "a+", rfp) == NULL)
+       fail("could not reopen %s: %s", rpath, strerror(errno));
+    setlinebuf(rfp);
```

Comment:
1) For what purpose do you think this is needed?
2) These lines must be protected by an `if (rfp)` guard.
3) Please see line 273 for the preferred form for calling freopen().
4) I'm not sure if the additional setlinebuf() call is needed: either it is unnecessary here, or it is also needed after line 273.

Reply:
1. Reassign the file descriptor to the actual context point.
2. I think it's not necessary, because if there is no rfp object, then there will be an error message and there is an error handler, and if there is an object, then the buffer will work.
3. line 273 (272 ???) have other target (fflush(stdout)).

Reply:
1. I don't understand what you mean by that. Normally one would use freopen() only if a) you want to open a different path, or b) you want to open with a different mode. But neither of those are changing here, so it should be a null operation.
2. It is legitimate to run pcoul without a logfile - I do this all the time. When that is the case, rfp will be NULL and should not be written to.
3. Sorry, not sure where I got that line number from, it should be 720: https://github.com/hvds/divrep/blob/1877d057621c934a90f37578af5226624b3eb1e8/coul.c#L720